lib/encoding/json.toit

```diff
@@ -10,16 +10,36 @@ MAX_BUFFER_GROWTH_ ::= 1024
 
 /**
 Encodes the $obj as a JSON ByteArray.
-The $obj must be null or an instance of int, bool, float, string, List, or Map.
-  Maps must have only string keys.  The elements of lists and the values of
+The $obj must be a supported type, which means either a type supported
+  by the converter block or an instance of int, bool, float, string, List
+  or Map.
+Maps must have only string keys.  The elements of lists and the values of
   maps can be any of the above supported types.
+The converter block is passed an object to be serialized and an instance
+  of the $Encoder class.  If it returns a non-null value, that value will
+  be serialized instead of the object that was passed in.  Alternatively,
+  the converter block can call the encode or put_unquoted methods on the
+  encoder.
 Utf-8 encoding is used for strings.
 */
-encode obj -> ByteArray:
+encode obj [converter] -> ByteArray:
```

Comment:
not sure `converter` is a good name.
In dart is is called `to_encodeable`, but the block here does more.
`on_non_json` or something like that would be more accurate.